src/backend/XmlHttpRequest.ts

```diff
@@ -390,16 +415,6 @@ export default class XmlHttpRequest extends BaseFileSystem implements FileSystem
   }
 
   /**
-   * Asynchronously download the given file.
-   */
-  private _requestFileAsync(p: string, type: 'buffer', cb: BFSCallback<Buffer>): void;
-  private _requestFileAsync(p: string, type: 'json', cb: BFSCallback<any>): void;
-  private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void;
-  private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void {
-    asyncDownloadFile(this.getXhrPath(p), type, cb);
-  }
-
-  /**
    * Synchronously download the given file.
    */
   private _requestFileSync(p: string, type: 'buffer'): Buffer;
```

Comment:
Another thing you could choose to tackle, or I can do post-merge: make these into properties, and replace with stubs that throw an [`ENOTSUP`](https://github.com/jvilk/BrowserFS/blob/master/src/core/file_system.ts#L757) ApiError when XHR isn't available.

As it stands, calling a synchronous method on this file system in e.g. Node will try to call XHR and will fail with a more mysterious error.